recipes/gtk/all/conanfile.py

```diff
@@ -49,21 +51,40 @@ def _build_subfolder(self):
     @property
     def _gtk4(self):
         return tools.Version("4.0.0") <= tools.Version(self.version) < tools.Version("5.0.0")
-    @property
 
+    @property
     def _gtk3(self):
         return tools.Version("3.0.0") <= tools.Version(self.version) < tools.Version("4.0.0")
 
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+            # Fix duplicate definitions of DllMain
+            self.options["gdk-pixbuf"].shared = True
+            # Fix segmentation fault
+            self.options["cairo"].shared = True
+            # The upstream meson file does not create a static library
+            self.options.shared = True
         if self.settings.os != "Linux":
             del self.options.with_wayland
             del self.options.with_x11
 
     def validate(self):
         if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
             raise ConanInvalidConfiguration("this recipes does not support GCC before version 5. contributions are welcome")
+        if str(self.settings.compiler) in ["Visual Studio", "msvc"]:
+            if tools.Version(self.version) < "4.3":
+                raise ConanInvalidConfiguration("MSVC support of this recipe requires at least gtk/4.3")
+            if not self.options["gdk-pixbuf"].shared:
+                raise ConanInvalidConfiguration("MSVC build requires shared gdk-pixbuf")
+            if not self.options["cairo"].shared:
+                raise ConanInvalidConfiguration("MSVC build requires shared cairo")
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("MSVC build supports shared gtk only")
```

Comment:
```suggestion
        if tools.Version(self.version) >= "4.1.0":
            if not self.options.shared:
                raise ConanInvalidConfiguration("gtk supports only shared since 4.1.0")
```
cf https://github.com/GNOME/gtk/commit/14f0a0addb9a195bad2f8651f93b95450b186bd6

Reply:
Shall we also set the default option to shared?